### DISCUSSION TITLE
Can I start `bin/vite dev` on a host other than localhost and with a self-signed SSL cert?

### DISCUSSION THREAD

**Discussion by jasonfb** [ElMassimo/vite_ruby#401] [Ideas]
I am developing an app where I need a local SSL (self-signed) certificate. 

The web server boots this way:

`bin/rails s -b 'ssl://0.0.0.0:3000?key=config/ssl/abc.local.key&cert=config/ssl/abc.local.crt'`

(so I can access the site locally at `https://abc.local:3000`)

Can I start Vite at `https://abc.local:3036`? 

the problem with current my setup is that the vite tag helpers pull in relative paths, which are looking for vite dev at https://abc.local:3036/ but it isn't running there (what I have in the Procfile.dev for vite is the standard `bin/vite dev`)

![DevTools - momeas local:3000:rooms:b5b9b55c-fe56-40e6-8e47-eab30c75fb48 2023-09-04 10-56-01](https://github.com/ElMassimo/vite_ruby/assets/59002/e29d3059-604e-4cdb-8cc2-25b2168226d6)

**Comment by jasonfb** [comment]
I tried this in my config/vite.json file but it seem like the settings are ignored

```
{
  "all": {
    "sourceCodeDir": "app/frontend",
    "watchAdditionalPaths": []
  },
  "development": {
    "autoBuild": true,
    "port": "3035",
    "host": "abc.local",
    "publicOutputDir": "vite-dev",
    "https:": true
  },
  "test": {
    "autoBuild": true,
    "publicOutputDir": "vite-test",
    "port": 3037
  }
}
```


when I run `bin/vite dev`, I get:
![Screenshot 2023-09-04 at 11 07 59 AM](https://github.com/ElMassimo/vite_ruby/assets/59002/6a6378e1-1b0e-4dd9-8f98-cb7fac26f2dc)


Shouldn't I be getting 

`https://abc.local:3035` 
based on the above settings?


also, I have already generated the local SSL key & certs, so I need to know how to tell Vite to use them when serving the assets or else it won't know how to terminal the connect on the self-signed SSL domain


Alternatively,  is there an easier way to do this? Specifically, can I just tell `vite_typescript_tag` to be loading assets from `localhost:3036` (and then I guess deal with the CORS)?

I suppose either way is ok to develop with.

**Reply by jasonfb** [reply]
what was happening here is that i had mistakenly overwritten vite.config.ts with some test settings and that was making it so that my changes in `config/vite.json` weren't being picked up. (I guess this requires the `vite.config.ts` to have the RubyPlugin to work correctly)

**Comment by jasonfb** [comment]
I got this working with these settings:

`vite.config.ts`
```
import { defineConfig } from 'vite'
import RubyPlugin from 'vite-plugin-ruby'

export default defineConfig({
  plugins: [
    RubyPlugin(),
  ],
  server: {
    https: {
      "key": "config/ssl/abc.local.key",
      "cert": "config/ssl/abc.local.crt"
    },
    "port": 3036,
    "host": "abc.local",
  }
})
```

`config/vite.json`
```
{
  "all": {
    "sourceCodeDir": "app/frontend",
    "watchAdditionalPths": []
  },
  "development": {
    "host": "abc.local",
    "https": true,
    "autoBuild": true,
    "publicOutputDir": "vite-dev"
  },
  "test": {
    "autoBuild": true,
    "publicOutputDir": "vite-test",
    "port": 3038
  }
}

```



Notice that 1) it was necessary to configure host in `config/vite.json`, but it was impossible to configure the SSL cert & key in that file.
Instead, I was unable to get the SSL cert set from the RUBY config, as the https form of taking an object with two keys (key and cert) only when configured in `vite.config.ts` (at root of repo)

now the vite dev server boots up as expected, and the corresponding web server produces URLS to this vite dev server appropriately
![vite with ssl](https://github.com/ElMassimo/vite_ruby/assets/59002/71b709c9-5335-451b-bd98-ee32644d4270)

**Reply by kyleroberthammond** [reply]
Thank you! This was bothering me for a few hours!
